ride-cleansing/src/main/java/org/apache/flink/training/exercises/ridecleansing/RideCleansingKeyByExercise.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.training.exercises.ridecleansing;
+
+import org.apache.flink.api.common.JobExecutionResult;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.PrintSinkFunction;
+import org.apache.flink.streaming.api.functions.sink.SinkFunction;
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+import org.apache.flink.training.exercises.common.datatypes.TaxiFare;
+import org.apache.flink.training.exercises.common.sources.TaxiFareGenerator;
+
+/**
+ * The Ride Cleansing exercise from the Flink training.
+ *
+ * <p>The task of this exercise is to filter a data stream of taxi ride records to keep only rides
+ * that both start and end within New York City. The resulting stream should be printed.
+ */
+public class RideCleansingKeyByExercise {
```

Comment:
Sorry, I don't understand purpose of this class.
Is it something additional from your side which you decided to try?

Reply:
yes this is an additional one

Reply:
ok